### PR TITLE
consolidate consts to prune dependencies to k/k

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind cleanup
/priority important-soon

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: part of #81172

**Special notes for your reviewer**:

/assign @andrewsykim 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/cloud/node_controller.go [hunk 1]
```diff
@@ -36,11 +36,26 @@ import (
...
+	// the cloud-controller-manager
+	// originated from https://github.com/kubernetes/kubernetes/blob/35cf6b6cbc/pkg/kubelet/apis/well_known_annotations.go#L25
+	annotationProvidedIPAddr = "alpha.kubernetes.io/provided-node-ip"
+
+	// taintExternalCloudProvider sets this taint on a node to mark it as unusable,
+	// when kubelet is started with the "external" cloud provider, until a controller
+	// from the cloud-controller-manager intitializes this node, and then removes
+	// the taint
+	// originated from https://github.com/kubernetes/kubernetes/blob/35cf6b6cbc/pkg/scheduler/api/well_known_labels.go#L64
+	taintExternalCloudProvider = "node.cloudprovider.kubernetes.io/uninitialized"
```
Comment (andrewsykim) [inline pkg/controller/cloud/node_controller.go]: Maybe move this to `k8s.io/cloud-provider` instead and replace remaining references to point to that too?

File: pkg/controller/cloud/node_controller.go [hunk 2]
```diff
@@ -36,11 +36,26 @@ import (
 	clientretry "k8s.io/client-go/util/retry"
 	cloudprovider "k8s.io/cloud-provider"
 	"k8s.io/klog"
-	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
-	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
```
Comment (liggitt) [inline pkg/controller/cloud/node_controller.go]: see discussion on moving the external-facing pieces from this package to a consumable location in https://github.com/kubernetes/kubernetes/issues/82390

Comment (Huang-Wei) [reply pkg/controller/cloud/node_controller.go]: @liggitt do you have a concrete idea of where to hold this kind of const strings? For example, under apimachinery path?

**Comment by andrewsykim** [comment]
/close

Looks like a dupe of https://github.com/kubernetes/kubernetes/pull/84364

**Pull request closed without merging**